notebooks/instructlab-knowledge/subset-selection.ipynb

```diff
@@ -0,0 +1,165 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import os\n",
+    "\n",
+    "chunk_lookup = {}\n",
+    "with open('data/chunks.jsonl', encoding='utf-8') as fin:\n",
+    "    os.makedirs('tmp', exist_ok=True)\n",
+    "    for line in fin.readlines():\n",
+    "        chunk_json = json.loads(line)\n",
+    "        chunk_lookup[chunk_json['chunk']] = chunk_json\n",
+    "\n",
+    "print(f'Read {len(chunk_lookup)} chunks')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "058dcbc4-097d-412f-a8b6-18ce4ee844fc",
+   "metadata": {},
+   "source": [
+    "# Preparation for performing subset selection\n",
+    "\n",
+    "We check out the "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0594d7bb-228b-468a-99a0-38c14651a562",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!git clone git@github.com:krishnatejakk/DataCurate4LLMs.git\n",
+    "%cd DataCurate4LLMs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2b2c8b60-daa9-48e4-b680-0b59588609ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "\n",
+    "seed = random.randint(0,10000)\n",
+    "\n",
+    "config = f\"\"\"{{\n",
+    "    \"output_dir\": \"output\",\n",
+    "    \"encoder_model\": \"BAAI/bge-large-en-v1.5\",\n",
+    "    \"encoder_type\": \"bge\",\n",
+    "    \"instruction\": \"Generate embeddings that capture the semantic meaning of text segments across multiple domains, ensuring g\\\n",
+    "eneralization and suitability for clustering based on semantic similarity.\",\n",
+    "    \"query_description\": \"default\",\n",
+    "    \"templates\": {{\n",
+    "        \"default\": \"{{{{ chunk }}}}\"\n",
+    "    }},\n",
+    "    \"template_name\": \"default\",\n",
+    "    \"num_folds\": 1,\n",
+    "    \"num_gpus\": 1,\n",
+    "    \"subset_sizes\": [\"5\"],\n",
+    "    \"epsilon\": 0.01,\n",
+    "    \"seed\": {seed}\n",
+    "}}\"\"\"\n",
+    "\n",
+    "config_path = 'subset_config.json'\n",
+    "with open(config_path, 'w') as f:\n",
+    "    f.write(config)\n",
+    "\n",
+    "print(config)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7efc49a-c08d-42f9-ad7a-b0723b49982f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!sed -i -e 's ^faiss-gpu$ faiss-gpu-cu12 g' requirements.txt # fix faiss dependency; yes you can use spaces as delimiters for sed expressions\n",
+    "!pip install -qq -r requirements.txt\n",
+    "!pip install -qq submodlib-py"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b086790-5a9f-4e2f-b130-2d9f9f8a0319",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python data_subset_selection.py --input_files '../data/chunks.jsonl' --output_dir '../data' --config '../subset_config.json' --retry_delay 1 --subset_sizes 5 --num_gpus 1"
+   ]
+  },
+  {
```

Comment:
The intention of this cell is just to filter the chunks that were selected out from all of the original chunks to make sure all of the chunks are in the same format as `chunks.jsonl` right?

Reply:
Yes, added some text explaining this.